Subject: patches/loopback-revert.patch


revert this commit:

commit 58f539740b1ccfc5ef4e509ec2efe82621b546e3
Author: Eric Dumazet <dada1@cosmosbay.com>
Date:   Fri Oct 20 00:32:41 2006 -0700

    [NET]: Can use __get_cpu_var() instead of per_cpu() in loopback driver.

    As BHs are off in loopback_xmit(), preemption cannot occurs, so we can
    use __get_cpu_var() instead of per_cpu() (and avoid a
    preempt_enable()/preempt_disable() pair)

    Signed-off-by: Eric Dumazet <dada1@cosmosbay.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Ingo Molnar <mingo@elte.hu>
---
 drivers/net/loopback.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

Index: linux-2.6-tip/drivers/net/loopback.c
===================================================================
--- linux-2.6-tip.orig/drivers/net/loopback.c
+++ linux-2.6-tip/drivers/net/loopback.c
@@ -76,11 +76,11 @@ static int loopback_xmit(struct sk_buff 
 
 	skb->protocol = eth_type_trans(skb,dev);
 
-	/* it's OK to use per_cpu_ptr() because BHs are off */
 	pcpu_lstats = dev->ml_priv;
-	lb_stats = per_cpu_ptr(pcpu_lstats, smp_processor_id());
+	lb_stats = per_cpu_ptr(pcpu_lstats, get_cpu());
 	lb_stats->bytes += skb->len;
 	lb_stats->packets++;
+	put_cpu();
 
 	netif_rx(skb);
 
